packages/smooth_app/lib/pages/product/product_page.dart

```diff
@@ -375,6 +375,16 @@ class _ProductPageState extends State<ProductPage> {
           i < _product.categoriesTags!.length;
           i++) {
         final String categoryTag = _product.categoriesTags![i];
+        final String currentLanguageCode =
```

Comment:
Shall we do this outside of the loop?

Reply:
If we need it elsewhere we could move it outside. It would be great to let users override the language in the preferences in fact.

Reply:
+1, I love to demo the app to people in their native language whenever possible

Reply:
I agree with @slava-sh about moving the code about `currentLanguageCode` just before the `i` loop. But not very important.
I would have reused `currentLanguageCode` a bit further as the `languageCode` parameter for `CategoryProductQuery`. Not very important either.

Reply:
OK we can do that. Why do we have a for loop by the way? It seems to just check one element.

Reply:
@stephanegigandet You're right, we don't need a loop. I guess that once upon a time that made sense to display all the hierarchy, but currently we only care about the finest level.
Getting rid of the `for` and replacing it with `final int i = _product.categoriesTags!.length - 1;` would definitely simplify the code.